code_samples/logging.js

```diff
@@ -0,0 +1,79 @@
+var winston = require('winston');
+var Config = require('../.').Config;
+var HazelcastClient = require('../.').Client;
+
+if(process.argv.length != 3){
+    console.log('You must give exactly one argument to run this tool!!!');
+}else{
+    switch (process.argv[2]){
+       
+        case 'winstonLogging.js':
+            var winstonAdapter = {
+                logger: new (winston.Logger)({
+                    transports: [
+                        new (winston.transports.Console)()
+                    ]
+                }),
+
+                levels: [
+                    'error',
+                    'warn',
+                    'info',
+                    'debug',
+                    'silly'
+                ],
+
+                log: function(level, className, message, furtherInfo) {
+                    this.logger.log(this.levels[level], className + ' ' + message);
+                }
+            };
+
+            var loggingHappened = false;
+
+
+            winstonAdapter.logger.on('logging', function(transport, level, msg, meta) {
+                loggingHappened = true;
+            });
+
+            //winston logging is active in case of winston object given as property
+            var cfg = new Config.ClientConfig();
+            cfg.properties['hazelcast.logging'] = winstonAdapter;
```

Comment:
I think this sample is unneccessarily repetitive. What you can do is to put only `cfg.properties['hazelcast.logging'] = logType` in switch-case. The rest of the code is the same for all samples.